src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/SubjectAccessRequestTest.kt

```diff
@@ -0,0 +1,298 @@
+    assertThat(actual)
+      .hasStatus(FORBIDDEN.value())
+      .hasUserMessage("Access Denied")
+      .hasDeveloperMessage("Access denied on uri=/subject-access-request")
+  }
+
+  @Test
+  fun `should return 204 if no content found`() {
+    webTestClient.get()
+      .uri { uriBuilder: UriBuilder ->
+        uriBuilder
+          .path(URI_TEMPLATE)
+          .queryParam("prn", aValidPrisonNumber())
+          .build()
+      }.bearerToken(
+        buildAccessToken(
+          roles = listOf("ROLE_SAR_DATA_ACCESS"),
+          privateKey = keyPair.private,
+        ),
+      )
+      .exchange()
+      .expectStatus()
+      .isNoContent
+      .expectBody().isEmpty
+  }
+
+  @Test
+  fun `should return 209 error if called with crn param`() {
+    webTestClient.get()
+      .uri { uriBuilder: UriBuilder ->
+        uriBuilder
+          .path(URI_TEMPLATE)
+          .queryParam("crn", aValidPrisonNumber())
+          .build()
+      }.bearerToken(
+        buildAccessToken(
+          roles = listOf("ROLE_SAR_DATA_ACCESS"),
+          privateKey = keyPair.private,
+        ),
+      )
+      .exchange()
+      .expectStatus()
+      .isEqualTo(209)
+      .expectBody().isEmpty
+  }
+
+  @Test
+  fun `should return 400 error if called without prn or crn param`() {
+    val response = webTestClient.get()
+      .uri { uriBuilder: UriBuilder ->
+        uriBuilder
+          .path(URI_TEMPLATE)
+          .build()
+      }.bearerToken(
+        buildAccessToken(
+          roles = listOf("ROLE_SAR_DATA_ACCESS"),
+          privateKey = keyPair.private,
+        ),
+      )
+      .exchange()
+      .expectStatus()
+      .isBadRequest
+      .returnResult(ErrorResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual)
+      .hasStatus(BAD_REQUEST.value())
+      .hasUserMessage("One of prn or crn must be supplied.")
+      .hasDeveloperMessage("One of prn or crn must be supplied.")
+  }
+
+  @Test
+  fun `should get induction and goals for specific prisoner without date filtering`() {
+    // Given
+    val prisonNumbers = listOf(aValidPrisonNumber(), anotherValidPrisonNumber())
+    prisonNumbers.map {
+      createInduction(
+        prisonNumber = it,
+        createInductionRequest = aValidCreateInductionRequestForPrisonerLookingToWork(),
+      )
+      createActionPlan(
+        prisonNumber = it,
+        createActionPlanRequest = aValidCreateActionPlanRequest(
+          goals = listOf(aValidCreateGoalRequest(title = "Goal 1")),
+        ),
+      )
+    }
+
+    // When
+    val response = webTestClient.sarRequest(aValidPrisonNumber(), null, null)
+
+    // Then
+    val responseBody = response
+      .expectStatus().isOk
+      .expectBody(HmppsSubjectAccessRequestContent::class.java)
+      .returnResult().responseBody
+
+    val content = objectMapper.convertValue(responseBody!!.content, SubjectAccessRequestContent::class.java)
```

Comment:
Nice 👍 
I like it that we are using the real target types, rather than reproducing them in the tests. Nice 👍